### PR TITLE
[AJER][OPS-3993] Initial build

### PR DESCRIPTION
**Pull request by andrewrichards1**

### CODE REVIEW

File: app/model/StatusTypes.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+sealed abstract class StatusType extends EnumEntry {
+}
+
+object StatusTypes extends Enum[StatusType] {
+
+  def forCode(code: String): Option[StatusType] = values.find(_.toString == code)
+
+  case object SENT extends StatusType {
```
Comment (pawelpanasewicz) [inline app/model/StatusTypes.scala]: Status

File: test/controllers/TpsControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import repository.TpsRepo
+import support.{AuthWireMockResponses, ItSpec, TestConnector, TpsData}
+
+class TpsControllerSpec extends ItSpec {
+
+  val repo = injector.instanceOf[TpsRepo]
+  val testConnector = injector.instanceOf[TestConnector]
+
+  override def beforeEach(): Unit = {
+    val remove = repo.removeAll().futureValue
```
Comment (pawelpanasewicz) [inline test/controllers/TpsControllerSpec.scala:30]: super.beforeEach()

File: test/model/StatusTypesSpec.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+import play.api.libs.json.{JsString, Json}
+import support.{RichMatchers, UnitSpec}
+
+class StatusTypesSpec extends UnitSpec with RichMatchers {
```
Comment (pawelpanasewicz) [inline test/model/StatusTypesSpec.scala]: remove RichMatchers, it' s in UnitSpec

File: test/repository/TpsRepoSpec.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import play.api.libs.json.Json
+import reactivemongo.api.commands.UpdateWriteResult
+import support.{ItSpec, TpsData}
+
+class TpsRepoSpec extends ItSpec {
+
+  val repo = injector.instanceOf[TpsRepo]
+
+  override def beforeEach(): Unit = {
+    val remove = repo.removeAll().futureValue
```
Comment (pawelpanasewicz) [inline test/repository/TpsRepoSpec.scala:29]: super.befreEach

File: test/support/AuthWireMockResponses.scala [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      )
+    )
+  }
+
+  def authorised(authProvider: String, strideUserId: String): StubMapping = {
+    stubFor(post(urlEqualTo("/auth/authorise"))
+      .withRequestBody(
+        equalToJson(
+          s"""
+             |{
```
Comment (pawelpanasewicz) [inline test/support/AuthWireMockResponses.scala]: get rid of pipes and give intellij hint about json

File: test/support/AuthWireMockResponses.scala [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                     |}
+       """.stripMargin)))
+
+  }
+
+  def failsWith(error: String): StubMapping = {
+    stubFor(post(urlEqualTo("/auth/authorise"))
+      .willReturn(aResponse()
+        .withStatus(401)
+        .withHeader("WWW-Authenticate", s"""MDTP detail=\"$error\"""")
```
Comment (pawelpanasewicz) [inline test/support/AuthWireMockResponses.scala]: escape is not needed

File: test/support/ItSpec.scala [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  extends FreeSpecLike
+  with RichMatchers
+  with BeforeAndAfterEach
+  with GuiceOneServerPerTest
+  with WireMockSupport
+  with Matchers {
+
+  implicit lazy val ec: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  lazy val servicesConfig = fakeApplication.injector.instanceOf[ServicesConfig]
```
Comment (pawelpanasewicz) [inline test/support/ItSpec.scala:66]: these probably should be `def`s

File: test/support/ItSpec.scala [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  lazy val config = fakeApplication.injector.instanceOf[Configuration]
```
Comment (pawelpanasewicz) [inline test/support/ItSpec.scala:67]: replace `fakeApplication` with `app`

File: test/support/ItSpec.scala [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  override def fakeApplication(): Application = new GuiceApplicationBuilder()
+    .overrides(GuiceableModule.fromGuiceModules(Seq(overridingsModule)))
+    .configure(configMap).build()
+
+  def configMap = Map[String, Any](
+    "mongodb.uri " -> "mongodb://localhost:27017/tps-payments-it",
+    "microservice.services.auth.port" -> WireMockSupport.port
+  )
+
+  def injector: Injector = fakeApplication().injector
```
Comment (pawelpanasewicz) [inline test/support/ItSpec.scala:95]: app.injector

File: test/support/TpsData.scala [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.time.LocalDateTime
+
+import model._
+import play.api.libs.json.{JsValue, Json}
+import reactivemongo.bson.BSONObjectID
+
+object TpsData {
+
+  val created = LocalDateTime.now
```
Comment (pawelpanasewicz) [inline test/support/TpsData.scala]: now(clock)

**Review by pawelpanasewicz — APPROVED** [review]

File: test/support/ItSpec.scala (hunk 7 above)
Comment (andrewrichards1) [reply test/support/ItSpec.scala:66]: cannot have lazy and defs ... seems to be working fine

File: test/support/ItSpec.scala (hunk 8 above)
Comment (andrewrichards1) [reply test/support/ItSpec.scala:67]: does not work

**Pull request merged**